src/system/file_path.cc

```diff
@@ -40,7 +40,7 @@ FilePath::FilePath(string file_path, const  FileType ft)
     if (ft == input_file) {
     	if ( FilePath::is_absolute_path(abs_file_path_) ) {
     	} else {
-            abs_file_path_ = root_dir + DIR_DELIMITER + abs_file_path_;
+            abs_file_path_ = input_root_dir + DIR_DELIMITER + abs_file_path_;
     	}
     } else if (ft == output_file) {
         if ( FilePath::is_absolute_path(abs_file_path_) ) {
```

Comment:
Possibly use boost to separate the file name and check the match parent_type() == output_dir.